### PR TITLE
content-sources: update migration repos script add repair images_repos.

### PR DESCRIPTION
**Pull request by ldjebran**
# Description
Repair images_repos images with repos that has duplicate urls, left only one repo with that url and remove the others.

FIXES: THEEDGE-3262

## Type of change

What is it?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [X] Tests update
- [ ] Refactor

# Checklist:

- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] I run `make pre-commit` to check fmt/vet/lint/test-no-fdo

### CODE REVIEW

**Review by loadtheaccumulator — COMMENTED** [review]
Just a couple of suggestions. Otherwise, LGTM

File: cmd/migraterepos/main.go [hunk 1]
```diff
@@ -68,7 +68,10 @@ func main() {
 			handleExist(err)
 			return
 		}
-
+		if err := repairrepos.RepairDuplicateImagesReposURLS(); err != nil {
+			handleExist(err)
```
[the remark this answers is not in the document]
Comment (ldjebran) [reply cmd/migraterepos/main.go]: function renamed to cleanupAndExit

File: cmd/migraterepos/repairrepos/repairrepos.go [hunk 2]
```diff
@@ -65,6 +65,89 @@ func RepairUrls() (int, error) {
...
+		log.WithFields(log.Fields{"URL": url, "image_id": imageID, "to-remove": len(otherReposIDS), "error": err.Error()}).Info("error occurred  while deleting image repos with duplicates urls in image_repos")
+		return err
+	}
+
+	log.WithFields(log.Fields{"URL": url, "image_id": imageID, "removed": len(otherReposIDS)}).Info("... repair images_repos url duplicates starting finished")
+
+	return nil
+}
+
+func RepairDuplicateImagesReposURLS() error {
```
[the remark this answers is not in the document]
Comment (ldjebran) [reply cmd/migraterepos/repairrepos/repairrepos.go:103]: add missing comments for public functions. seems the linter does not verify if there is a comment for public functions

**Comment by ldjebran** [comment]
@loadtheaccumulator comments addressed please revisit.

File: cmd/migraterepos/repairrepos/repairrepos.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (loadtheaccumulator) [reply cmd/migraterepos/repairrepos/repairrepos.go:103]: I have it enabled in my IDE, so maybe something we can add to the PR linter in the future.

**Review by loadtheaccumulator — APPROVED** [review]
lgtm;

File: cmd/migraterepos/repairrepos/repairrepos.go [hunk 3]
```diff
@@ -65,6 +65,91 @@ func RepairUrls() (int, error) {
 	return affectedURLS, nil
 }
 
+func repairDuplicateImagesReposURLS(imageID uint, url string) error {
```
Comment (mgold1234) [inline cmd/migraterepos/repairrepos/repairrepos.go:68]: missing the // repairDuplicateImagesReposURLS comment ,, as Jonathan mentioned

File: cmd/migraterepos/repairrepos/repairrepos.go [hunk 4]
```diff
@@ -65,6 +65,91 @@ func RepairUrls() (int, error) {
...
+
+		imagesUrls += len(imagesReposData)
+		page++
+	}
+
+	log.WithField("images_repos_urls", imagesUrls).Info("... repair images_repos urls duplicates repair finished")
+
+	return nil
+}
+
```
Comment (mgold1234) [inline cmd/migraterepos/repairrepos/repairrepos.go:152]: here as well

File: cmd/migraterepos/repairrepos/repairrepos.go (hunk 3 above)
Comment (ldjebran) [reply cmd/migraterepos/repairrepos/repairrepos.go:68]: @mgold1234 we comment Public functions only

**Review by mgold1234 — APPROVED** [review]
/lgtm

**Comment by ldjebran** [comment]
remove "do not merge" and enable migrate feature flag on stage

**Pull request merged**